src/services/google-analytics/loadGoogleAnalytics.ts

```diff
@@ -49,6 +49,12 @@ const loadGoogleAnalytics = (trackerId: string) => {
 // until the real google analytics object is ready to replace it
 const createGAShim = (trackerId: string) => {
   window.gtag = function () {
+    // Cleanup the dataLayer array after every 100 events
+    if (window.dataLayer.length > 100) {
+      window.dataLayer.push(function (this: any) {
+        this.reset();
+      });
+    }
```

Comment:
1) The function should not take `this` as an argument
2) Is this any better than linking the cleanup of the data layer to the navigation from one app to another?

Reply:
I found the part of the gtag script that prevents the infinite growth of the queue:

![image](https://user-images.githubusercontent.com/6834224/179321869-c43e8883-c457-4a12-903d-8498f25e2f89.png)

Gtag allows the queue to grow up to a thousand elements, and then for every new push, removes the oldest element(s) from the queue so that its size never exceeds 1000 elements. I don't know why they allow the queue to get to 1000; but in any case, I now have proof that the queue will not grow indefinitely.

Let's delete our cleanup code then. Sorry I brought this up.

Reply:
Thanks for sharing that. It's good to get these checked anyways.